### PR TITLE
feat(Banner): new banner design

### PR DESCRIPTION
**Pull request by matthprost**
## Type

- Feature

### ✨ New banner design

The design of the Banner has a new look including patterns as background for both promotional and intro banners.

## Relevant logs and/or screenshots

Screenshot 2023-07-25 at 16 21 51

### CODE REVIEW

File: packages/ui/src/components/Banner/index.tsx [hunk 1]
```diff
@@ -1,27 +1,88 @@
...
+      background-image: url(${promotionCompactLeftPattern}),
+        url(${promotionCompactRightPattern}),
+        ${theme.colors.other.gradients.background.purple};
+      background-position: left, right;
+      background-repeat: no-repeat, no-repeat;
+      background-size: contain, contain;
+    `
+  }
+
+  return css``
```
Comment (chambo-e) [inline packages/ui/src/components/Banner/index.tsx:73]:
```suggestion
  return null
```

Does not work ?

File: packages/ui/src/components/Banner/index.tsx [hunk 2]
```diff
@@ -1,27 +1,88 @@
...
+
+  if (variant === 'promotional' && size === 'medium') {
+    return css`
+      background-image: url(${promotionPattern}),
+        ${theme.colors.other.gradients.background.purple};
+      background-position: right;
+      background-repeat: no-repeat;
+      background-size: contain;
+    `
+  }
```
Comment (chambo-e) [inline packages/ui/src/components/Banner/index.tsx:49]:
```ts
if (size === 'small') {
  if (intro)
  
  if (promo
}

if (size === 'medium)' {
  ...
}
```

Easier to understand

File: packages/ui/src/components/Banner/index.tsx [hunk 3]
```diff
@@ -59,13 +126,18 @@ export const Banner = ({
...
+
   const [opened, setOpened] = useState(true)
 
   if (!opened) return null
 
   return (
     <Container variant={variant} size={size} className={className}>
-      {image}
+      <ImageStack size={size} justifyContent="center">
+        {image ?? <img src={defaultImage} alt="" />}
```
Comment (chambo-e) [inline packages/ui/src/components/Banner/index.tsx:139]:
```suggestion
        {image ? <img src={defaultImage} alt="" /> : null}
```

Also should be 
```suggestion
        {image ? <img src={image ?? defaultImage} alt="" /> : null}
```

no  ?

**Comment by QuiiBz** [comment]
> ### ⚠️ No Changeset found
> Latest commit: [b81ebf9](https://github.com/scaleway/ultraviolet/commit/b81ebf98c74d0130479506999df9b0b45b8b9b5d)
> 
> Merging this PR will not cause a version bump for any packages. If these changes should not result in a new version, you're good to go. **If these changes should result in a version bump, you need to add a changeset.**
> 
> This PR includes no changesets
> [Click here to learn what changesets are, and how to add one](https://github.com/changesets/changesets/blob/main/docs/adding-a-changeset.md?rgh-link-date=2023-07-24T09%3A19%3A37Z).
> 
> [Click here if you're a maintainer who wants to add a changeset to this PR](https://github.com/scaleway/ultraviolet/new/feat/enhance-banner?filename=.changeset/clean-oranges-smell.md&value=---%0A%22%40ultraviolet%2Fui%22%3A%20patch%0A---%0A%0Afeat(Banner)%3A%20new%20banner%20design%0A)

@matthprost missing changeset

File: packages/ui/src/components/Banner/index.tsx (hunk 3 above)
Comment (chambo-e) [reply packages/ui/src/components/Banner/index.tsx]: @matthprost So what's the point of passing `image` if we are always going to show the defaultImage ?

**Review by fabienhebert — APPROVED** [review]

File: packages/ui/src/components/Banner/index.tsx (hunk 3 above)
Comment (chambo-e) [reply packages/ui/src/components/Banner/index.tsx:143]: Alright, that's me I'm dumb ... my bad

**Review by chambo-e — APPROVED** [review]

**Review by PetmePant — APPROVED** [review]

**Pull request merged**